### PR TITLE
[DATA-802] [DATA-803] Add button to select, edit emitted details

### PR DESCRIPTION
**Pull request by alexis-wei**

### CODE REVIEW

File: src/elements/select/select.svelte [hunk 1]
```diff
@@ -241,6 +251,10 @@ const handleClearAll = () => {
   }
 };
 
+const handleButtonClick = () => {
+  dispatch('buttonclick');
```
Comment (sbal13) [inline src/elements/select/select.svelte:255]: nit: typically we do multi-word events with a dash like 'button-click'

File: src/stories/select.stories.mdx [hunk 2]
```diff
@@ -33,9 +33,24 @@ import gpioOptions from './assets/gpio-options'
...
+      description: 'Whether a button exists at the bottom to conduct an action',
+      table: { defaultValue: { summary: 'false' } }
+    },
+    buttontext: {
+      description: 'The text on the button',
+      table: { defaultValue: { summary: 'ENTER' } }
+    },
+    buttonicon: {
+      description: 'The icon on the button, can use any from v-icon',
+      table: { defaultValue: { summary: '' } }
```
Comment (sbal13) [inline src/stories/select.stories.mdx:46]: might be nice to have an input for this for the Canvas view, just need to add `control: 'text'`. I'd say the same goes for buttontext

File: src/stories/select.stories.mdx [hunk 3]
```diff
@@ -83,6 +98,29 @@ For user triggered actions
   </Story>
 </Canvas>
 
+<Canvas>
+  <Story
+    name='Photos'
```
Comment (sbal13) [inline src/stories/select.stories.mdx:103]: Change this name to something more descriptive of what you're showcasing, maybe like "With Button" or something

File: src/elements/select/select.svelte [hunk 4]
```diff
@@ -25,6 +25,9 @@ export let tooltip = '';
 export let state: 'info' | 'warn' | 'error' | '' = 'info';
 export let showpill = 'false';
 export let clearable = 'true';
+export let button = 'false';
```
Comment (sbal13) [inline src/elements/select/select.svelte:28]: Can we change this name to something like "withbutton" or something? Just feels more obvious that it's a boolean value than "button"

**Review by sbal13 — CHANGES_REQUESTED** [review]
Looks good! Just a few naming things and quality of life additions to the Canvas

File: src/elements/select/select.svelte (hunk 1 above)
Comment (alexis-wei) [reply src/elements/select/select.svelte:255]: okie!

File: src/stories/select.stories.mdx (hunk 2 above)
Comment (alexis-wei) [reply src/stories/select.stories.mdx]: added!

File: src/stories/select.stories.mdx (hunk 3 above)
Comment (alexis-wei) [reply src/stories/select.stories.mdx]: yup

File: src/elements/select/select.svelte (hunk 4 above)
Comment (alexis-wei) [reply src/elements/select/select.svelte]: yup

**Pull request merged**